### ISSUE TITLE
"F" Course Grade Appearing When in Cat-Only and Only Uncategorized Items Contain Scores

### ISSUE THREAD

**Issue by kyleblythe** [sakaiproject/sakai#1485]
To reproduce:

1) Set up a GB with "Categories Only" in place
2) Add items to categories
3) Add uncategorized items
4) Enter scores ONLY for one (or more) of the uncategorized items

This results in the "null" course grade value being replaced with an "F" for the student(s) in question. Since uncategorized items are not part of course grade calculations, the "null" value should persist.

**Comment by steveswinsburg** [comment]
Two issues here. 
1. was a possible NPE that I have fixed (above commit).
2. is that we have told the UI to display an F for blank course grades.

**Comment by steveswinsburg** [comment]
@kyleblythe I spent a bit of time trying to trace down the issue here. There doesn't appear to be a way to resolve this in the existing service code so that a 0.0 grade doesn't get mapped to an F grade as the determination of this isn't aware that the grade value may have come from uncategorised assignments only.

However, a while back I introduced the new API method:
`public org.sakaiproject.service.gradebook.shared.CourseGrade getCourseGradeForStudent(String gradebookUid, String userUuid) {`
which is what we use for the dynamic course grade calculations and is free of this bug. I could extend this to cater for all students in the course and then we could swap out our use of the older API method to this one.
